pkgs/applications/editors/kakoune/wrapper.nix

```diff
@@ -14,10 +14,17 @@ in
     paths = [ kakoune ] ++ requestedPlugins;
 
     postBuild = ''
+      # create a directory for bins that kakoune needs
+      # access to, without polluting the users path by adding
+      # that binary nested with this symlinkJoin.
+      mkdir -p $out/share/kak/bin
+
       # location of kak binary is used to find ../share/kak/autoload,
       # unless explicitly overriden with KAKOUNE_RUNTIME
       rm "$out/bin/kak"
-      makeWrapper "${kakoune}/bin/kak" "$out/bin/kak" --set KAKOUNE_RUNTIME "$out/share/kak"
+      makeWrapper "${kakoune}/bin/kak" "$out/bin/kak" \
+        --set KAKOUNE_RUNTIME "$out/share/kak" \
```

Comment:
https://github.com/mawww/kakoune/blob/master/src/main.cc#L202-L214

Reply:
Good pointer. I tested and it seems that if i put hop-kak binary into that directory, it actually picks it up without a kak/share directory. I'm going to keep it where it is though w/ the path modification, because that directory may get polluted if a user installs many binary-based plugins. Either way, this is a user-invisible change, so if this turns out to be a bad decision we can revert it in the future without too much hassle.